src/pyrenew/process/differencedprocess.py

```diff
@@ -198,15 +193,23 @@ def sample(
             Whose value entry is a single array representing the
             undifferenced timeseries
         """
-        diffs, *_ = self.fundamental_process.sample(
-            *args,
-            n=(n - self.differencing_order),
-            init_vals=fundamental_process_init_vals,
-            **kwargs,
-        )
+        if n < 1:
+            raise ValueError("n must be positive. " f"Got {n}")
+
+        n_diffs = n - self.differencing_order
+        if n_diffs > 0:
+            diff_samp, *_ = self.fundamental_process.sample(
+                *args,
+                n=n_diffs,
+                init_vals=fundamental_process_init_vals,
+                **kwargs,
+            )
+            diffs = diff_samp.value
+        else:
+            diffs = jnp.array([])
         return (
             SampledValue(
-                value=self.integrate(init_vals, diffs.value),
+                value=self.integrate(init_vals, diffs)[:n],
```

Comment:
slight preference to compute self.integrate and store it to make the return statement more readable.

Reply:
addressed in 755fc39